### DISCUSSION TITLE
Cannot use `x-bind:x-if`

### DISCUSSION THREAD

**Discussion by Sleepful** [alpinejs/alpine#3265] [7. Bugs (archived)]
Basically...
This will always display
```html
            <template x-if="true">
              <span>hi</span>
            </template>
```
This will never display:
```html
            <template x-bind:x-if="true">
              <span>hi</span>
            </template>
```
using `    "alpinejs": "^3.10.5",`

**Comment by Sleepful** [comment]
`x-bind:x-ref` also does not work and `x-ref` does not work inside a `x-for`, probably related to https://github.com/alpinejs/alpine/discussions/1527

**Reply by ekwoka** [reply]
What is the use case for doing `x-bind:x-ref` etc?

When using `x-bind` to assign an object, there x-ref and x-if will work.

**Reply by Sleepful** [reply]
no, `x-bind:x-ref` does *not* work, the thread I linked has lots of people talking about this, there's even a link to the official docs saying that on v3 binding `x-ref` dynamically no longer works.

**Reply by Sleepful** [reply]
@ekwoka the use-case for something like `x-bind:x-ref` is to be able to attach unique `x-ref`s to dynamically created elements, for instance the elements created through `x-for`. Currently you can control the behavior of `x-for` and its child tags by keeping track of the array that it binds to or by using more manual `element.querySelector` javascript.

**Reply by ekwoka** [reply]
Yeah, but why would you need that?

x-ref is very rarely even useful in a well architected application.

Probably the least useful if all the built in directives, aside from x-html

**Comment by Sleepful** [comment]
`x-bind:x-on:click` also does not work, currently finding work-arounds like:

```
@click="remove_new($el.getAttribute('ref'))" x-bind:ref="`new_${i}`
```

For anyone looking, another good workaround mentioned in this other thread: https://github.com/alpinejs/alpine/discussions/1527#discussioncomment-2717277
